### PR TITLE
DDF-1468 changed alias method to pull from system property to avoid potential race condition with config admin

### PR DESCRIPTION
**Pull request by stustison**
@lessarderic 
@coyotesqrl 
@pklinef

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/226)

### CODE REVIEW

**Comment by coyotesqrl** [comment]
I guess that might be _just a smidge_ more efficient. :) :+1:

File: platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/Security.java [hunk 1]
```diff
@@ -219,23 +216,7 @@ public static PKIAuthenticationTokenFactory createPKITokenFactory() {
...
-            } catch (IOException e) {
-                LOGGER.error("Could not get reference to configuration admin. ", e);
-                return null;
-            }
-            Dictionary<String, Object> properties = config.getProperties();
-            return (String) properties.get("host");
-        }
-        LOGGER.warn("Unable to get system certificate alias");
-        return null;
+        return System.getProperty("org.codice.ddf.system.hostname");
```
Comment (clockard) [inline platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/Security.java]: Might want to use the constant SystemBaseUrl.HOST

**Comment by clockard** [comment]
:+1:

**Comment by shaundmorris** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/Security.java, line 218 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/226#-K-_QjIh7Exfsbo-YKoZ)** ([raw file](https://github.com/codice/ddf/blob/4d26fe19420aabeb588fd499f4a8325b80a9164d/platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/Security.java#L218)):</sup>
While here, should we refactor the method name to "getCertificateAlias" ? Missing the last "e"


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/226)

**Comment by lessarderic** [comment]
:+1: 


---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/Security.java, line 218 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/226#-K-_QjIh7Exfsbo-YKoZ)** ([raw file](https://github.com/codice/ddf/blob/4d26fe19420aabeb588fd499f4a8325b80a9164d/platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/Security.java#L218)):</sup>
+1


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/226)

**Comment by coyotesqrl** [comment]
I just duplicated these test failures locally. There is a problem with this change. I guess none of the branches with null returns in the old implementation were ever hit.

**Comment by clockard** [comment]
Looks like we still need to update the unit tests for this. Need to set the hostname system property.

**Comment by coyotesqrl** [comment]
That's good to know. I'm happy it's just a test environment setup issue.

**Comment by stustison** [comment]
I forgot to commit the unit test. It should be gtg now.

File: platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/Security.java (hunk 1 above)
Comment (stustison) [inline platform/security/core/security-core-impl/src/main/java/ddf/security/common/util/Security.java]: I'm going to leave that for the other ticket. I didn't want to introduce any new dependencies with this change.

**Comment by pklinef** [comment]
:+1:

**Comment by coyotesqrl** [comment]
:+1:

**Comment by clockard** [comment]
:+1:

**Pull request merged**